### PR TITLE
Tasks 1-8: Create basic endpoints

### PR DESCRIPTION
**Pull request by Disembow**
# This PR covers next tasks:

## Task-1: Print "Hello World!" to the console

**Acceptance criteria**:

- Create `index.js` file created.
- `hello world` is printed in console by running application.

## Task-2: Make first server to server API request

**Acceptance criteria**:

- `package.json` file and project configuration is initialized
- Add start command to package.json
- API request library is chosen and installed
- NASA API token is created by sign up (https://api.nasa.gov/index.html)
- NASA API endpoint is called and response is received (GET https://api.nasa.gov/neo/rest/v1/feed?start_date=START_DATE&end_date=END_DATE&api_key=API_KEY)
- Response in JSON format is printed to the console

## Task-3: Add environment variables

**Acceptance criteria**:

- Add `.env` for environment variables
- `.env` should be added to .gitignore, and .env.example should be created to store template keys
- Import the token and api link into our `index.js` file
- Start the application using variables from the environment file

## Task-4: Connect Express framework

**Acceptance criteria**:

- Connect framework
- Start server on port `4000`
- Print to the console that the server is running and on what port
- Create a separate configuration file to manage environment variables

## Task-5: Create a first endpoint

**Acceptance criteria**:

- Add endpoint for the example `"/meteors"`
- Output the response in JSON format to endpoint response
- Test response with browser

## Task-6: Explore the Postman tool

**Acceptance criteria**:

- Install Postman
- Make multiple requests to the NASA API
- Make a request to your proxy server

## Task-7: Add logic to your app

**Acceptance criteria**:

- Add logic to the base endpoint `/meteors`

## Task-8: Refactor code

**Acceptance criteria**:

- Divide the application into layers
- Add the controllers/services/repositories layers
- Split logic by layers
- Add custom Exception. It should include a status code and an error message. Error handler should be able to work with it
- Create middleware for error handling. This middleware should accept an error and return a user-friendly response to the client (e.g., error status and message). Ensure that all errors from the all layers are passed to the error handler
- Add “page not found” endpoint. For now it can return “Page not found” string

### CODE REVIEW

File: app/config/config.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import dotenv from "dotenv";
+
+dotenv.config();
+
+const config = {
+  BASE_API_URL: "https://api.nasa.gov",
+  PORT: process.env.PORT || 4000,
+  API_KEY: process.env.API_KEY,
+  START_DATE: "2024-11-17",
+  END_DATE: "2024-11-18",
```
Comment (mdziarkach) [inline app/config/config.js]: Put startDate and endDate inside of getMeteors controller

File: app/repositories/meteors.repository.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import axios from "axios";
+import { getApiUrl } from "../utils/getApiUrl.js";
+
+export const getMeteors = async (startDate, endDate) => {
+  const url = getApiUrl(startDate, endDate);
```
Comment (mdziarkach) [inline app/repositories/meteors.repository.js]: you can form url right here

File: app/repositories/meteors.repository.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+
+  const response = await axios.get(url);
```
Comment (mdziarkach) [inline app/repositories/meteors.repository.js:10]: use params for forming query parameters 

https://stackoverflow.com/a/53501339/12225394

File: app/utils/errorHandler.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+export const errorHandler = (err, req, res, next) => {
+  console.error(err.stack);
+
+  res.status(err.status || 500).json({
+    success: false,
+    message: err.message || "Internal Server Error",
+    stack: err.stack,
```
Comment (mdziarkach) [inline app/utils/errorHandler.js]: user shouldn't know about stack
this information only for developers

File: app/repositories/meteors.repository.js (hunk 2 above)
Comment (Disembow) [reply app/repositories/meteors.repository.js]: Sorry, didn't catch your idea. Do you mean I shouldn't form url right here?

File: app/repositories/meteors.repository.js (hunk 3 above)
Comment (Disembow) [reply app/repositories/meteors.repository.js:10]: Am I mistaken, or adding query params is the very next task [#9](https://godelonline.sharepoint.com/Functions/LearningAndDevelopment/SitePages/Node---Proxy-Server.aspx?csf=1&web=1&e=Z6bZS6&OR=Teams-HL&CT=1732611317249&clickparams=eyJBcHBOYW1lIjoiVGVhbXMtRGVza3RvcCIsIkFwcFZlcnNpb24iOiI0OS8yNDEwMjAwMTMxOCIsIkhhc0ZlZGVyYXRlZFVzZXIiOmZhbHNlfQ%3D%3D&CID=5a831324-1a88-4207-96aa-41cca15bce8e#task-9-add-parameters-to-the-query)?

File: app/utils/errorHandler.js (hunk 4 above)
Comment (Disembow) [reply app/utils/errorHandler.js]: Thank you, I'll rework it with next commit

File: app/repositories/meteors.repository.js (hunk 2 above)
Comment (mdziarkach) [reply app/repositories/meteors.repository.js]: yes

File: app/repositories/meteors.repository.js (hunk 3 above)
Comment (mdziarkach) [reply app/repositories/meteors.repository.js:10]: but you already added them

Even if it’s in the next task, it’s already in the code, so I can leave a comment :)

Comment (Disembow) [reply app/repositories/meteors.repository.js:10]: if you don’t mind I'll make these changes in the next PR

**Pull request merged**